pynucastro/eos/fermi_integrals.py

```diff
@@ -0,0 +1,203 @@
+
+                fval = kernel
+                df_deta = fval
+                df_deta2 = fval
+                df_dbeta = fval*xsq*denom2i
+                df_dbeta2 = -df_dbeta*xsq*denom2i
+                df_deta_dbeta = df_dbeta
+        else:
+            if ((x-eta) < 2.0e2):
+                xdk = x**k
+                sqrt_factor = np.sqrt(1 + (x*beta/2))
+                exp_factor = np.exp(x - eta)
+                denom = exp_factor + 1
+                denomi = 1/denom
+                kernel = xdk*sqrt_factor*denomi
+                denom2i =  1/(4 + 2*beta*x)
+
+                fval = kernel
+                df_deta = fval*exp_factor*denomi
+                df_deta2 = (2*exp_factor*denomi - 1)*df_deta
+                df_dbeta = fval*x*denom2i
+                df_dbeta2 = -df_dbeta*x*denom2i
+                df_deta_dbeta = df_dbeta*exp_factor*denomi
+            else:
+                xdk = x**k
+                sqrt_factor = np.sqrt(1 + (x*beta/2))
+                denomi = np.exp(eta - x)
+                kernel = xdk*sqrt_factor*denomi
+                denom2i =  1/(4 + 2*beta*x)
+
+                fval = kernel
+                df_deta = fval
+                df_deta2 = fval
+                df_dbeta = fval*x*denom2i
+                df_dbeta2 = -df_dbeta*x*denom2i
+                df_deta_dbeta = df_dbeta
+
+        if(eta_der == 0 and beta_der == 0):
+            return fval
+        elif(eta_der == 1 and beta_der == 0):
+            return df_deta
+        elif(eta_der == 2 and beta_der == 0):
+            return df_deta2
+        elif(eta_der == 0 and beta_der == 1):
+            return df_dbeta
+        elif(eta_der == 0 and beta_der == 2):
+            return df_dbeta2
+        elif(eta_der == 1 and beta_der == 1):
+            return df_deta_dbeta
+        else:
+            raise ValueError("Invalid derivative order")
+
+    def _fermi_points(self, eta_der=0):
+
+        eta = self.eta
+
+        D = [3.3609, 4.99551, 3.93830, 4.17444]
+        sigma = [9.1186e-2, 9.11856e-2, 9.11856e-2, 9.11856e-2]
+
+        xi = [(1/s)*np.log(1+np.exp(s*(eta-d))) for d, s in zip(D,sigma)]
+
+        a1 = np.array([6.7774, 6.77740, 6.77740, 6.77740])
+        b1 = np.array([1.1418, 1.14180, 1.14180, 1.14180])
+        c1 = np.array([2.9826, 2.98255, 2.98255, 2.98255])
+
+        a2 = np.array([3.7601, 3.76010, 3.76010, 3.76010])
+        b2 = np.array([9.3719e-2, 9.37188e-2, 9.37188e-2, 9.37188e-2])
+        c2 = np.array([2.1064e-2, 2.10635e-2, 2.10635e-2, 2.10635e-2])
+        d2 = np.array([3.1084e1, 3.95015e1, 3.14499e1, 3.05412e1])
+        e2 = np.array([1.0056, 1.00557, 1.00557, 1.00557])
+
+        a3 = np.array([7.5669, 7.56690, 7.56690, 7.56690])
+        b3 = np.array([1.1695, 1.16953, 1.16953, 1.16953])
+        c3 = np.array([7.5416e-1, 7.54162, 7.54162, 7.54162])
+        d3 = np.array([6.6559, 7.64734, 6.86346, 7.88030])
+        e3 = np.array([-1.2819, -1.28190e-1, -1.28190e-1, -1.28190e-1])
```

Comment:
the values in the source code are not the same as in the paper for e3 -- they are all -1.28190e-1

Reply:
the code also has more precision for the first column